src/cbmc/cbmc_solvers.cpp

```diff
@@ -189,8 +187,6 @@ class cbmc_solver_with_filet:public cbmc_solverst::solvert
   ~cbmc_solver_with_filet()
   {
     // delete the prop before the file
-    delete prop_conv_ptr;
```

Comment:
Shouldn't solvert own all the objects? The comment says: "The solver class (that takes care of allocated objects)" - which appears to be far from true. My suggestion is for solvert to take a second argument `propt *prop` that my be a nullptr. solvert should then take care of deleting all non-null members. Otherwise it's rather impossible to get the order of cleanup right (as seems to be proven).